### PR TITLE
switch GCR -> Artifact-Registry

### PR DESCRIPTION
**Pull request by ccwienk**
GCR has been deprecated [0] in favour of Artifact-Registry.

Thus, change push-targets for OCI-Images:

- europe-docker.pkg.dev/gardener-project/snapshots for snapshots
- europe-docker.pkg.dev/gardener-project/releases for releases
- europe-docker.pkg.dev/gardener-project/public for combined view of snapshots + releases

[0]
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

**Release note**:

```breaking operator
Change OCI Image Registry from GCR (`eu.gcr.io/gardener-project`) to Artifact-Registry (`europe-docker.pkg.dev/gardener-project/releases`). Users should update their references.

```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
This PR proposes changes that would break the pipeline definition:
```
<invalid YAML>: found duplicate anchor 'mcmpa-image'; first occurrence
  in "<unicode string>", line 28, column 54:
     ... ontroller-manager-provider-aws: &mcmpa-image
                                         ^
second occurrence
  in "<unicode string>", line 75, column 54:
     ... ontroller-manager-provider-aws: &mcmpa-image
                                         ^
```

**Comment by gardener-robot-ci-1** [comment]
This PR proposes changes that would break the pipeline definition:
```
<invalid YAML>: found duplicate anchor 'mcmpa-image'; first occurrence
  in "<unicode string>", line 29, column 54:
     ... ontroller-manager-provider-aws: &mcmpa-image
                                         ^
second occurrence
  in "<unicode string>", line 76, column 54:
     ... ontroller-manager-provider-aws: &mcmpa-image
                                         ^
```

**Comment by gardener-robot-ci-1** [comment]
The pipeline-definition has been fixed.

File: .test-defs/IntegrationTest.yaml [hunk 1]
```diff
@@ -1,14 +0,0 @@
-kind: TestDefinition
```
[the remark this answers is not in the document]
Comment (hendrikKahl) [reply .test-defs/IntegrationTest.yaml:1]: The content of any `TestDefinition` is repository specific and includes "local" domain knowledge. Therefore it should to be updated by the maintainers.

[the remark this answers is not in the document]
Comment (ccwienk) [reply .test-defs/IntegrationTest.yaml:1]: I will take this as a "maybe it should better stay". -> will remove removal from this pull-request, and simply leave it as-is.

**Review by aaronfern — APPROVED** [review]
/lgtm

**Pull request merged**